### PR TITLE
Removed name property from BaseOperator

### PR DESCRIPTION
**Pull request by ilooner**
This needs to be merged with MLHR-1795

### CODE REVIEW

**Comment by 243826** [comment]
By virtue of removing something from the API, this pull request can only be raised against v4.0.0. Please submit it there.

**Comment by tweise** [comment]
Let's deprecate it and remove all usage.

**Comment by vrozov** [comment]
The change to engine/pom.xml is independent from API change. Can you port it to 3.1.0 as a separate pull or as part of deprecate changes.

File: common/src/main/java/com/datatorrent/common/util/BaseOperator.java [hunk 1]
```diff
@@ -28,26 +28,6 @@
  */
 public class BaseOperator implements Operator
 {
-  private String name;
```
Comment (ilooner) [inline common/src/main/java/com/datatorrent/common/util/BaseOperator.java]: deprecate

**Pull request closed without merging**